### PR TITLE
feat(suite-native): add device PIN protection card for non-T1B1 models

### PR DESCRIPTION
**Pull request by yanascz**
Support for T1B1 to be added in a follow-up PR.

## Related Issues

#14615 #14616

## Screenshots:

Simulator Screenshot - iPhone 15 Pro - 2024-10-16 at 14 57 07
Simulator Screenshot - iPhone 15 Pro - 2024-10-16 at 14 57 13
Simulator Screenshot - iPhone 15 Pro - 2024-10-16 at 14 57 24
Simulator Screenshot - iPhone 15 Pro - 2024-10-16 at 15 56 19

### CODE REVIEW

File: suite-native/device-manager/src/components/DeviceSettingsButton.tsx [hunk 1]
```diff
@@ -27,26 +27,29 @@ type DeviceInfoButtonProps = {
...
     const navigation = useNavigation<NavigationProp>();
     const { setIsDeviceManagerVisible } = useDeviceManager();
     const selectedDevice = useSelector(selectDevice);
 
     const handleDeviceRedirect = () => {
         setIsDeviceManagerVisible(false);
-        navigation.navigate(RootStackRoutes.DeviceInfo);
-        analytics.report({ type: EventType.DeviceManagerClick, payload: { action: 'deviceInfo' } });
+        navigation.navigate(RootStackRoutes.DeviceStack);
+        analytics.report({ type: EventType.DeviceManagerClick, payload: { action: 'deviceInfo' } }); // TODO: Rename to deviceSettings?
```
Comment (yanascz) [inline suite-native/device-manager/src/components/DeviceSettingsButton.tsx]: Is it safe to just rename this one? Are there any other events we wanna report?

File: suite-native/device/src/navigation/DevicePinProtectionStackNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
```
Comment (Nodonisko) [inline suite-native/device/src/navigation/DevicePinProtectionStackNavigator.tsx]: please create new package `@suite-native/module-device-settings`, packages without `module` prefixes shoudn't navigations/screens

Comment (yanascz) [reply suite-native/device/src/navigation/DevicePinProtectionStackNavigator.tsx]: There is already navigator and screen in the `@suite-native/device` package. 🤷‍♂️

Comment (Nodonisko) [reply suite-native/device/src/navigation/DevicePinProtectionStackNavigator.tsx]: Yea, we should move it too :D

File: suite-native/device-manager/src/components/DeviceSettingsButton.tsx (hunk 1 above)
Comment (matejkriz) [reply suite-native/device-manager/src/components/DeviceSettingsButton.tsx]: Yes, we can rename it, just mention it in the Notion in [Changelog](https://www.notion.so/satoshilabs/Changelog-Mobile-app-946307159a294bcdb90b94e90eefd292?pvs=4) and [Overview](https://www.notion.so/satoshilabs/da354c80a23a44a2814748968f701c5a?v=6a0a9e88bd04494db46c139336122d37&pvs=4) 🙏 

And regarding other events, I would copy desktop which means reporting PIN change and PIN protection change.

File: suite-native/module-device-settings/src/components/DevicePinProtectionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    if (!device) {
+        return;
+    }
+
+    return (
+        <DeviceSettingsCard
+            icon="password"
+            title={<Translation id="moduleDeviceSettings.pinProtection.title" />}
+        >
+            <VStack marginTop="sp2" spacing="sp16">
```
Comment (matejkriz) [inline suite-native/module-device-settings/src/components/DevicePinProtectionCard.tsx:23]: Right padding seems to be off, it supposed to be 16, same as on the bottom but it's less.
image

**Review by matejkriz — APPROVED** [review]
Except the one small issue with the spacing it seems to be working very well and smooth! I can't break it even the unhappy paths, canceling, locked devices...

File: suite-native/module-device-settings/src/screens/ContinueOnTrezorScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+import { selectDeviceModel } from '@suite-common/wallet-core';
+import { Box, Text } from '@suite-native/atoms';
+import { ConnectorImage, DeviceImage } from '@suite-native/device';
+import { Translation } from '@suite-native/intl';
+import { Screen, ScreenSubHeader } from '@suite-native/navigation';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import TrezorConnect from '@trezor/connect';
+
+const SCREEN_HEIGHT = Dimensions.get('screen').height;
```
Comment (matejkriz) [inline suite-native/module-device-settings/src/screens/ContinueOnTrezorScreen.tsx]: super nitpick, ignore right away 😆 :  I would prefer using util `getScreenHeight`

Comment (yanascz) [reply suite-native/module-device-settings/src/screens/ContinueOnTrezorScreen.tsx]: Fixed.

Comment (matejkriz) [reply suite-native/module-device-settings/src/screens/ContinueOnTrezorScreen.tsx]: You can't import `/lib` folder, please fix the import `import { getScreenHeight } from '@trezor/env-utils';` 
Linter is also complaining about that so I hope it's possible to setup auto import feature to never use /lib folder.

Comment (yanascz) [reply suite-native/module-device-settings/src/screens/ContinueOnTrezorScreen.tsx]: My bad, I haven't noticed.

File: suite-native/module-device-settings/src/components/DevicePinProtectionCard.tsx (hunk 3 above)
Comment (matejkriz) [reply suite-native/module-device-settings/src/components/DevicePinProtectionCard.tsx:23]: This fixes the issue for me 06abe1fa9d5ff106ac8a2b6692d52e69ca8ad031

**Review by matejkriz — APPROVED** [review]

**Pull request merged**